### PR TITLE
Partial typehints for bleak/__init__.py

### PR DESCRIPTION
**Pull request by Siecje**

### CODE REVIEW

File: bleak/__init__.py [hunk 1]
```diff
@@ -154,11 +155,16 @@ def __init__(
             **kwargs,
         )
 
-    async def __aenter__(self):
+    async def __aenter__(self) -> "BleakScanner":
```
Comment (dlech) [inline bleak/__init__.py:158]: We should be able to drop the quotes since we have `from __future__ import annotations` in this file already.

File: bleak/__init__.py [hunk 2]
```diff
@@ -513,19 +519,21 @@ def __init__(
...
-            if services is None
-            else set(map(normalize_uuid_str, services)),
+            disconnected_callback=(
+                None
+                if disconnected_callback is None
+                else functools.partial(disconnected_callback, self)
+            ),
+            services=(
+                None if services is None else set(map(normalize_uuid_str, services))
+            ),
```
Comment (dlech) [inline bleak/__init__.py:536]: This looks unrelated. I'm guessing `black` did this?

File: bleak/__init__.py [hunk 3]
```diff
@@ -553,19 +559,24 @@ def mtu_size(self) -> int:
...
         return f"{self.__class__.__name__}, {self.address}"
 
-    def __repr__(self):
+    def __repr__(self) -> str:
         return f"<{self.__class__.__name__}, {self.address}, {type(self._backend)}>"
 
     # Async Context managers
 
-    async def __aenter__(self):
+    async def __aenter__(self) -> "BleakClient":
```
Comment (dlech) [inline bleak/__init__.py:570]: Can remove the quotes here too.

File: bleak/__init__.py (hunk 2 above)
Comment (Siecje) [reply bleak/__init__.py]: Yes black.

**Comment by dlech** [comment]
Merged, thanks!
